drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -527,9 +701,91 @@ local function occupancy_attr_handler(driver, device, ib, response)
   device:emit_event(ib.data.value == 0x01 and capabilities.motionSensor.motion.active() or capabilities.motionSensor.motion.inactive())
 end
 
+local function initial_press_event_handler(driver, device, ib, response)
+  if get_field_for_endpoint(device, SUPPORTS_MULTI_PRESS, ib.endpoint_id) then
+    -- Receipt of an InitialPress event means we do not want to ignore the next MultiPressComplete event
+    -- or else we would potentially not create the expected button capability event
+    set_field_for_endpoint(device, IGNORE_NEXT_MPC, ib.endpoint_id, nil)
+  else
+    if get_field_for_endpoint(device, INITIAL_PRESS_ONLY, ib.endpoint_id) then
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.button.button.pushed({state_change = true}))
+    elseif get_field_for_endpoint(device, EMULATE_HELD, ib.endpoint_id) then
+      -- if our button doesn't differentiate between short and long holds, do it in code by keeping track of the press down time
+      init_press(device, ib.endpoint_id)
+    end
+  end
+end
+
+-- if the device distinguishes a long press event, it will always be a "held"
+-- there's also a "long release" event, but this event is required to come first
+local function long_press_event_handler(driver, device, ib, response)
+  device:emit_event_for_endpoint(ib.endpoint_id, capabilities.button.button.held({state_change = true}))
+  if get_field_for_endpoint(device, SUPPORTS_MULTI_PRESS, ib.endpoint_id) then
+    -- Ignore the next MultiPressComplete event if it is sent as part of this "long press" event sequence
+    set_field_for_endpoint(device, IGNORE_NEXT_MPC, ib.endpoint_id, true)
+  end
+end
+
+local function short_release_event_handler(driver, device, ib, response)
+  if not get_field_for_endpoint(device, SUPPORTS_MULTI_PRESS, ib.endpoint_id) then
+    if get_field_for_endpoint(device, EMULATE_HELD, ib.endpoint_id) then
+      emulate_held_event(device, ib.endpoint_id)
+    else
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.button.button.pushed({state_change = true}))
+    end
+  end
+end
+
+local function multi_press_complete_event_handler(driver, device, ib, response)
+  -- in the case of multiple button presses
+  -- emit number of times, multiple presses have been completed
+  if ib.data and get_field_for_endpoint(device, IGNORE_NEXT_MPC, ib.endpoint_id) ~= true then
+    local press_value = ib.data.elements.total_number_of_presses_counted.value
+    --capability only supports up to 6 presses
+    if press_value < 7 then
```

Comment:
similar to above, I suggest doing:
```suggestion
    if press_value >= 7 then
        log.warn("more than 7 button presses. aborting")
        return
   end
```
and then continue with the function

Reply:
Or rather, have this be the log:
 log.info(string.format("Number of presses (%d) not supported by capability", press_value))

Reply:
Similar to a previous comment you suggested, `IGNORE_NEXT_MPC` needs to be set to nil at the end of the function in all cases, so while an early return would be nice I'm not sure it will work. I would prefer to keep this as-is for this reason.

Reply:
Alright, good point. Sounds good to me :+1: